docs/user_configuration_file.md

```diff
@@ -0,0 +1,53 @@
+# User Configuration File
+A file defined by users aimed at communicating specific user behavior is required to perform the various transaction processing steps inside of this application.
+
+The file format is YAML and its location can be passed as parameters to user requests requiring user-specific configuration. The YAML document fields are described below (also in YAML, so you get used to the user config format).
+
+```yaml
+InternalInternalTransferReferences:
+    - Description: References that appear in transfers between banks of the same user.
+      Type: List of Strings
+      Context: These references help detecting and removing internal transfers, which are no-op and add noise, therefore cleaning the transaction list.
+
+BankProcessingTimeInDays
+    - Description: The maximum amount of time, among all configured user banks, banks take to process transactions.
```

Comment:
The intention here is that the user communicates what is the time that his/her bank takes to process transaction. Currently this value is used as a time window analysis for internal transfer detection, but in the future it could be used in other use cases.

I am following the principle of decoupling "raw user information" vs "its uses inside the application". While I think it's important that the user knows for what this is used, I think as a configuration parameter it's best for it to represent an information about how banks behave.

Extra documentation is certainly needed to detail what the app does, but we shouldn't couple with the user configuration.

That said, I'm curious if you still think the name is ambiguous and what name would you suggest?

Reply:
I agree that it's better to use variable names that are familiar to the user's reality. I was concerned that it might be ambiguous because a bank could have several processes with different processing times, but I confess that I didn't pay much attention to the 'context' field, where you explain that it is related to the time window analysis for internal transfer detection.